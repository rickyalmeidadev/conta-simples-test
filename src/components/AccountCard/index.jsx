import React from 'react';
import { Link } from 'react-router-dom';

const AccountCard = ({ title, description, buttonText, link, test }) => (
  <article data-testid={test}>
    <h2>{title}</h2>
    <p>{description}</p>
    <Link to={link}>{buttonText}</Link>
  </article>
);

export default AccountCard;
