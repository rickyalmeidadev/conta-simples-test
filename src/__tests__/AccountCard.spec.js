import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import AccountCard from '../components/AccountCard';

afterEach(cleanup);

describe('<AccountCard />', () => {
  const mockCard = {
    title: 'Conta corrente',
    description: 'Consulte entradas e saÃ­das pelo seu estrato.',
    buttonText: 'Ver estrato',
    link: '/statement',
    test: 'statement',
  };

  it('renders correctly', () => {
    const { asFragment } = render(
      <Router>
        <AccountCard {...mockCard} />
      </Router>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the cards', () => {
    const { getByTestId, getByText } = render(
      <Router>
        <AccountCard {...mockCard} />
      </Router>,
    );

    expect(getByTestId(mockCard.test)).toBeDefined();
    expect(getByText(mockCard.title)).toBeDefined();
    expect(getByText(mockCard.description)).toBeDefined();
    expect(getByText(mockCard.buttonText)).toBeDefined();
  });
});
