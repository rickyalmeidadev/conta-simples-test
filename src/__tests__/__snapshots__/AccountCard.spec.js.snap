// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountCard /> renders correctly 1`] = `
<DocumentFragment>
  <article
    data-testid="statement"
  >
    <h2>
      Conta corrente
    </h2>
    <p>
      Consulte entradas e sa√≠das pelo seu estrato.
    </p>
    <a
      href="/statement"
    >
      Ver estrato
    </a>
  </article>
</DocumentFragment>
`;
